api/controllers/game/spectate.js

```diff
@@ -1,17 +1,37 @@
 module.exports = async function (req, res) {
   try {
     const { gameId } = req.body;
-    const game = await gameService.populateGame({ gameId });
+    const promiseGame = await gameService.populateGame({ gameId });
+    const promiseSpectator = await userService.findUser({ userId: req.session.usr });
 
+    const [game, spectator] = await Promise.all([promiseGame, promiseSpectator]);
```

Comment:
Doing these in parallel is great 👍🏼 

Considering you're only using these variables once to reset them, you could also do something like this
```
    const promiseGame = ;
    const promiseSpectator = await userService.findUser({ userId: req.session.usr });

    const [game, spectator] = await Promise.all([
        gameService.populateGame({ gameId }),
        userService.findUser({ userId: req.session.usr }),
    ]);
``` 
I think those are promises by default so you likely don't need the await keyword for the individual use cases, but test it out to see.

Reply:
I like this b/c it's shorter and the calls to the promise returning functions are already short and easy to read. I say we remove the `promiseGame` and `promiseSpectator` variables altogether and use 
```
    const [game, spectator] = await Promise.all([
        gameService.populateGame({ gameId }),
        userService.findUser({ userId: req.session.usr }),
    ]);
```
as suggested above